streams/src/test/java/org/apache/kafka/test/MockProcessor.java

```diff
@@ -30,13 +29,9 @@
 
 import static org.junit.Assert.assertEquals;
 
-@SuppressWarnings("WeakerAccess")
 public class MockProcessor<K, V> extends AbstractProcessor<K, V> {
 
     public final ArrayList<String> processed = new ArrayList<>();
-    public final ArrayList<K> processedKeys = new ArrayList<>();
-    public final ArrayList<V> processedValues = new ArrayList<>();
-    public final ArrayList<KeyValueTimestamp> processedWithTimestamps = new ArrayList<>();
```

Comment:
IMHO, it would be better to keep this one in favor over `processed` that does not provide type safety...

Reply:
I understand. The reason I chose to take it out is that this field `processedWithTimestamps` is only used in one test: KGroupedStreamImplTest and the key/value types are String and long, whose string representation did not introduce regression. I actually felt that for this single test it is simpler to go with `processed` only where `processed` typed changed from `String` to `ValueAndTimestamp<KeyValue>`. But that involves removing this `KeyValueTimestamp` class and refactoring lots of unit tests as well.

How about I just file a JIRA for this bigger cleanup and merge it as is?

Reply:
SGTM.

Reply:
Filed https://issues.apache.org/jira/browse/KAFKA-8450